test/unit/services/instanceAdapter_spec.js

```diff
@@ -22,11 +22,29 @@ describe("The InstanceAdapter class ", function () {
 
 			instances.createInstance()
 			.then(function (result) {
+				var failedInstance = new Instance({
```

Comment:
I still don't understand why this is needed for the simple create test case. Perhaps it will be more clear with the changes you reference below?